test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java

```diff
@@ -175,25 +176,24 @@ private static boolean isWindows() {
     return System.getProperty("os.name", "").toLowerCase(Locale.US).contains("win");
   }
 
-  public Optional<Version> getDseVersion() {
-    return DSE_ENABLEMENT ? Optional.of(VERSION) : Optional.empty();
+  public static boolean isDistributionOf(BackendType type) {
+    return DISTRIBUTION == type;
+  }
+
+  public static boolean isDistributionOf(BackendType type, VersionComparator comparator) {
+    return isDistributionOf(type)
+        && comparator.accept(getDistributionVersion(), getCassandraVersion());
+  }
```

Comment:
I like this more since it at least preserves the one-shot comparison of backend type + version but overall I would still argue for something built around the Optional type.  But this PR has been held up long enough as-is and I don't feel strongly enough about it to hold things up any longer so we'll roll with this.